tests/actions/QueuedOnyxUpdatesTest.ts

```diff
@@ -0,0 +1,171 @@
+import type {OnyxUpdate} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import {flushQueue, queueOnyxUpdates} from '@libs/actions/QueuedOnyxUpdates';
+import type {OnyxKey} from '@src/ONYXKEYS';
+import ONYXKEYS from '@src/ONYXKEYS';
+import waitForBatchedUpdates from '../utils/waitForBatchedUpdates';
+
+const queuedOnyxUpdates: OnyxUpdate[] = [
+    {key: ONYXKEYS.NVP_TRY_FOCUS_MODE, value: true, onyxMethod: 'merge'},
+    {key: ONYXKEYS.PREFERRED_THEME, value: 'system', onyxMethod: 'merge'},
+    {key: ONYXKEYS.NVP_PREFERRED_LOCALE, value: 'en', onyxMethod: 'merge'},
+    {
+        key: ONYXKEYS.SESSION,
+        value: {
+            accountID: 18748326,
+            authToken: 'testToken',
+            email: 'abcd+2342424224@gmail.com',
+            encryptedAuthToken: 'testEncryptedAuthToken',
+            loading: false,
+        },
+        onyxMethod: 'merge',
+    },
+    {key: ONYXKEYS.IS_LOADING_APP, value: false, onyxMethod: 'merge'},
+    {
+        key: ONYXKEYS.CREDENTIALS,
+        value: {
+            autoGeneratedLogin: '',
+            autoGeneratedPassword: '',
+            login: 'abcd+2342424224@gmail.com',
+        },
+        onyxMethod: 'merge',
+    },
+    {key: ONYXKEYS.IS_SIDEBAR_LOADED, value: true, onyxMethod: 'merge'},
+    {key: `${ONYXKEYS.COLLECTION.REPORT}2175919089355165`, value: {reportID: 'reportID'}, onyxMethod: 'merge'},
+    {
+        key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}2175919089355165`,
+        value: {
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            '4135522899867010163': {
+                reportActionID: '4135522899867010163',
+            },
+        },
+        onyxMethod: 'merge',
+    },
+];
+
+jest.mock('@src/CONFIG', () => ({
+    IS_TEST_ENV: false,
+}));
+
+function getOnyxUpdateValue<T>(key: string): T | undefined {
+    return queuedOnyxUpdates.find((item) => item.key === key)?.value as T | undefined;
+}
+
+async function testOnyxKeyValue<T>(key: OnyxKey): Promise<void> {
+    const expectedValue = getOnyxUpdateValue<T>(key);
+    return new Promise<void>((resolve) => {
+        const connection = Onyx.connect({
+            key,
+            waitForCollectionCallback: false,
+            callback: (value) => {
+                Onyx.disconnect(connection);
+
+                expect(value).toEqual(expectedValue);
+                resolve();
+            },
+        });
+    });
+}
+
+describe('actions/QueuedOnyxUpdates', () => {
+    beforeAll(() => {
+        Onyx.init({
+            keys: ONYXKEYS,
+        });
+    });
+
+    beforeEach(() => {
+        return Onyx.clear().then(waitForBatchedUpdates);
+    });
+
+    describe('flushQueue', () => {
+        it('should filter queued updates when currentAccountID is undefined', async () => {
```

Comment:
@huult Can we mimic being offline, clear the session, and then check if all the reports are cleared?

Reply:
@getusha This is another case, and this code handles the case you mentioned

https://github.com/Expensify/App/blob/d82b96e66aaf0456b17ffe66e1a36977c2c56324/src/libs/Navigation/AppNavigator/AuthScreens.tsx#L401

Reply:
@getusha what do you think?

Reply:
@huult do we need to make adjustments? we would've caught this bug if the tests were written for the case

Reply:
@getusha Because the case of Offline mode -> Clear Session is a separate case, it is not related to our changes, I think. In our case, it follows this sequence: Offline mode -> Online mode -> API executed -> Added data to queue in Onyx -> Clear session -> Report removed.”